connectors/confluent.mdx

```diff
@@ -194,3 +193,58 @@ CREATE TABLE orders (
   'source.read_mode' = 'read_committed'
 );
 ```
+
+## Schema Registry integration
+
+Arroyo supports the Confluent Schema Registry for both JSON and Avro schemas. To use the Schema Registry
+for a table, you must have configured the connection profile with the schema registry endpoint and
+credentials. Then, it will be available as an option when setting the schema for the table.
+
+How Arroyo handles schemas depends on whether the table is a source or a sink and the format.
+
+### Source tables
+
+For source tables, Arroyo will read the schema from the schema registry at the time the table is created.
+This JSON or Avro schema is converted into a SQL table schema, as if it had been defined directly in SQL.
+
+#### JSON
+
+Because SQL has a less flexible type system than JSON schema, not all JSON schema features are supported
+directly. For any fields that cannot be converted to a SQL type, Arroyo will create a column of type `JSON`
+and store the raw JSON value in that column. The JSON value can then be processed via SQL JSON functions.
+
+The schema is read from the schema registry at the time the table is created, which for tables created via the UI
+is when the table is saved; for tables created via SQL it is when the pipeline is started. The schema is
+cached for the lifetime of the table, so if the schema in the schema registry is updated or deleted, reads
+from the table will continue to use the original schema.
+
+#### Avro
+
+For Avro, there are some features that cannot be converted to SQL types:
+
+* Unions (aside from unions that are a single type and `null`, which are converted to `nullable` columns)
+* Maps
+* Arrays
+
+Fields with these types will be re-encoded as JSON and stored in a `JSON` column, in which form they
+can be processed via SQL JSON functions.
+
+At table creation time (when the table is saved in the UI or when the pipeline
+is started via SQL for SQL-defined tables), Arroyo will read the schema from the schema registry and
+convert it to a SQL schema. That schema is used as the "reader" schema for the lifetime of the table
+and determines what fields are available. If the schema in the schema registry is updated or deleted,
+reads from the table will continue to use the original schema.
+
+When using the schema registry, Arroyo will also look at the schema ID for each
+message, and if it encounters a new schema ID, it will fetch the schema from the
+schema registry and use it as the "writer" schema. Messages are then decoded
+using the writer schema and then evolved to conform to the reader schema. This
+ensures that so long as the schema in the schema registry is compatible with the
```

Comment:
This is a fragment:
"This ensures that so long as the schema in the schema registry is compatible with the reader schema."